In [21]:
import pandas as pd
import numpy as np 

df = pd.read_csv("train_data.csv")
df.head()

ID HealthServiceArea Gender                    Race TypeOfAdmission  \
0   1     New York City      F              Other Race         Newborn   
1   2     New York City      M  Black/African American         Newborn   
2   3     New York City      M              Other Race         Newborn   
3   4     New York City      F              Other Race         Newborn   
4   5     New York City      F              Other Race         Newborn   

   CCSProcedureCode  APRSeverityOfIllnessCode           PaymentTypology  \
0               228                         1                  Medicaid   
1               228                         1                  Medicaid   
2               220                         1  Private Health Insurance   
3                 0                         1  Private Health Insurance   
4               228                         1                  Medicaid   

   BirthWeight EmergencyDepartmentIndicator  AverageCostInCounty  \
0         3700                            N                 2611   
1         2900                            N                 3242   
2         3200                            N                 3155   
3         3300                            N                 3155   
4         2600                            N                 2611   

   AverageChargesInCounty  AverageCostInFacility  AverageChargesInFacility  \
0                    9227                   1751                      8951   
1                    8966                   3338                      6409   
2                   11381                   4980                      9323   
3                   11381                   5826                     15680   
4                    9227                   6000                     14344   

   AverageIncomeInZipCode  LengthOfStay  
0                      45             1  
1                      34             1  
2                      45             1  
3                      59             1  
4                      59             1

In [22]:
df.drop(columns = ['ID','HealthServiceArea'],inplace=True,axis=1)
df.head()

Gender                    Race TypeOfAdmission  CCSProcedureCode  \
0      F              Other Race         Newborn               228   
1      M  Black/African American         Newborn               228   
2      M              Other Race         Newborn               220   
3      F              Other Race         Newborn                 0   
4      F              Other Race         Newborn               228   

   APRSeverityOfIllnessCode           PaymentTypology  BirthWeight  \
0                         1                  Medicaid         3700   
1                         1                  Medicaid         2900   
2                         1  Private Health Insurance         3200   
3                         1  Private Health Insurance         3300   
4                         1                  Medicaid         2600   

  EmergencyDepartmentIndicator  AverageCostInCounty  AverageChargesInCounty  \
0                            N                 2611                    9227   
1                            N                 3242                    8966   
2                            N                 3155                   11381   
3                            N                 3155                   11381   
4                            N                 2611                    9227   

   AverageCostInFacility  AverageChargesInFacility  AverageIncomeInZipCode  \
0                   1751                      8951                      45   
1                   3338                      6409                      34   
2                   4980                      9323                      45   
3                   5826                     15680                      59   
4                   6000                     14344                      59   

   LengthOfStay  
0             1  
1             1  
2             1  
3             1  
4             1

Drop the columns ID and HeatlServiceArea

#EDA and Statistics

In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 59966 entries, 0 to 59965
Data columns (total 14 columns):
 #   Column                        Non-Null Count  Dtype 
---  ------                        --------------  ----- 
 0   Gender                        59966 non-null  object
 1   Race                          59966 non-null  object
 2   TypeOfAdmission               59966 non-null  object
 3   CCSProcedureCode              59966 non-null  int64 
 4   APRSeverityOfIllnessCode      59966 non-null  int64 
 5   PaymentTypology               59966 non-null  object
 6   BirthWeight                   59966 non-null  int64 
 7   EmergencyDepartmentIndicator  59966 non-null  object
 8   AverageCostInCounty           59966 non-null  int64 
 9   AverageChargesInCounty        59966 non-null  int64 
 10  AverageCostInFacility         59966 non-null  int64 
 11  AverageChargesInFacility      59966 non-null  int64 
 12  AverageIncomeInZipCode        59966 non-null  int64 
 13  LengthOfStay    

The data has  categorical columns :
<br> 1.Gender
<br> 2.Race
<br> 3.PaymentTypology
<br> 4.Emergency Depearment Indicator 
<br> <b> Even though the columns CCSProcedureCode and APRSeverityofIllnessCode are numeric, they seem to be categorical in nature which must be verified 

In [25]:
df.describe()

CCSProcedureCode  APRSeverityOfIllnessCode   BirthWeight  \
count      59966.000000              59966.000000  59966.000000   
mean         155.404229                  1.254594   3336.298903   
std           89.541978                  0.546207    446.244475   
min           -1.000000                  1.000000   2500.000000   
25%          115.000000                  1.000000   3000.000000   
50%          220.000000                  1.000000   3300.000000   
75%          228.000000                  1.000000   3600.000000   
max          231.000000                  4.000000   7500.000000   

       AverageCostInCounty  AverageChargesInCounty  AverageCostInFacility  \
count         59966.000000            59966.000000           59966.000000   
mean           2372.806690             7979.126922            2396.414318   
std             639.755096             3220.291347            1248.501189   
min             712.000000             1243.000000             457.000000   
25%            2041.000000             4620.000000            1551.000000   
50%            2533.000000             9227.000000            1967.000000   
75%            2785.000000            10644.000000            2895.000000   
max            3242.000000            11381.000000            8114.000000   

       AverageChargesInFacility  AverageIncomeInZipCode  LengthOfStay  
count              59966.000000            59966.000000  59966.000000  
mean                7958.472668               59.811143      2.538405  
std                 3859.294711               21.470170      1.171246  
min                 1120.000000               28.000000      1.000000  
25%                 4438.000000               45.000000      2.000000  
50%                 7945.000000               55.000000      2.000000  
75%                11619.000000               74.000000      3.000000  
max                18466.000000              115.000000     10.000000

Few observations from the description statistics:
    <br>1. The average LengthOfStay(LOS) is 2.53 and the min=1 and max=10
    <br>2. The average AverageCostInCounty is 2372 and AvergaeChargesInCounty is 7979 . whats the correlatio between these two ?
    <br><b> It would be intersting toobserve the correlation between these and the LengthOfStay(LOS)

In [35]:
col_values = df['LengthOfStay'].values
df.groupby(['LengthOfStay']).count()
#unique_val = pd.unique(col_values)
#print(sorted(unique_val))

Gender   Race  TypeOfAdmission  CCSProcedureCode  \
LengthOfStay                                                     
1               8895   8895             8895              8895   
2              25000  25000            25000             25000   
3              16000  16000            16000             16000   
4               7504   7504             7504              7504   
5               1342   1342             1342              1342   
6                557    557              557               557   
7                346    346              346               346   
8                145    145              145               145   
9                 97     97               97                97   
10                80     80               80                80   

              APRSeverityOfIllnessCode  PaymentTypology  BirthWeight  \
LengthOfStay                                                           
1                                 8895             8895         8895   
2                                25000            25000        25000   
3                                16000            16000        16000   
4                                 7504             7504         7504   
5                                 1342             1342         1342   
6                                  557              557          557   
7                                  346              346          346   
8                                  145              145          145   
9                                   97               97           97   
10                                  80               80           80   

              EmergencyDepartmentIndicator  AverageCostInCounty  \
LengthOfStay                                                      
1                                     8895                 8895   
2                                    25000                25000   
3                                    16000                16000   
4                                     7504                 7504   
5                                     1342                 1342   
6                                      557                  557   
7                                      346                  346   
8                                      145                  145   
9                                       97                   97   
10                                      80                   80   

              AverageChargesInCounty  AverageCostInFacility  \
LengthOfStay                                                  
1                               8895                   8895   
2                              25000                  25000   
3                              16000                  16000   
4                               7504                   7504   
5                               1342                   1342   
6                                557                    557   
7                                346                    346   
8                                145                    145   
9                                 97                     97   
10                                80                     80   

              AverageChargesInFacility  AverageIncomeInZipCode  
LengthOfStay                                                    
1                                 8895                    8895  
2                                25000                   25000  
3                                16000                   16000  
4                                 7504                    7504  
5                                 1342                    1342  
6                                  557                     557  
7                                  346                     346  
8                                  145                     145  
9                                   97                      97  
10                                  80                      80

The examples with < 4 days is 49895 and 10,071 >=4 days .
<br> 1. (<  4 days) => 83.36 %
<br> 2. (>=4 days) => 16.64 %



In [36]:
#So lets change the LengthOfStay 
def change_los(x):
    if x < 4:
        return 0
    return 1
df['LengthOfStay'] = df['LengthOfStay'].apply(change_los)

In [37]:
df.head(3)

Gender                    Race TypeOfAdmission  CCSProcedureCode  \
0      F              Other Race         Newborn               228   
1      M  Black/African American         Newborn               228   
2      M              Other Race         Newborn               220   

   APRSeverityOfIllnessCode           PaymentTypology  BirthWeight  \
0                         1                  Medicaid         3700   
1                         1                  Medicaid         2900   
2                         1  Private Health Insurance         3200   

  EmergencyDepartmentIndicator  AverageCostInCounty  AverageChargesInCounty  \
0                            N                 2611                    9227   
1                            N                 3242                    8966   
2                            N                 3155                   11381   

   AverageCostInFacility  AverageChargesInFacility  AverageIncomeInZipCode  \
0                   1751                      8951                      45   
1                   3338                      6409                      34   
2                   4980                      9323                      45   

   LengthOfStay  
0             0  
1             0  
2             0

In [38]:
col_values = df['LengthOfStay'].values
df.groupby(['LengthOfStay']).count()

Gender   Race  TypeOfAdmission  CCSProcedureCode  \
LengthOfStay                                                     
0              49895  49895            49895             49895   
1              10071  10071            10071             10071   

              APRSeverityOfIllnessCode  PaymentTypology  BirthWeight  \
LengthOfStay                                                           
0                                49895            49895        49895   
1                                10071            10071        10071   

              EmergencyDepartmentIndicator  AverageCostInCounty  \
LengthOfStay                                                      
0                                    49895                49895   
1                                    10071                10071   

              AverageChargesInCounty  AverageCostInFacility  \
LengthOfStay                                                  
0                              49895                  49895   
1                              10071                  10071   

              AverageChargesInFacility  AverageIncomeInZipCode  
LengthOfStay                                                    
0                                49895                   49895  
1                                10071                   10071

<b> So the above change was successful and this confirms our previous undesrtanding that the dat contains samples LOS < 4 

Lets now look at if there any NULL values 

In [41]:
df.isnull().any()

Gender                          False
Race                            False
TypeOfAdmission                 False
CCSProcedureCode                False
APRSeverityOfIllnessCode        False
PaymentTypology                 False
BirthWeight                     False
EmergencyDepartmentIndicator    False
AverageCostInCounty             False
AverageChargesInCounty          False
AverageCostInFacility           False
AverageChargesInFacility        False
AverageIncomeInZipCode          False
LengthOfStay                    False
dtype: bool

<b> So the dataset doesnot contain any NULL values 